### PR TITLE
Week 6: Bowling Challenge

### PR DESCRIPTION
**Pull request by JG075**

### CODE REVIEW

File: src/Game.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+function Game() {
+  this.frames = [[[]]];
```
Comment (maryrosecook) [inline src/Game.js]: Yes, the code might be clearer if a Frame object was used to model the individual frames.

File: src/Game.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  this.currentFrame = 0;
+}
+
+Game.prototype.addScore = function(score) {
+  if (this.currentFrame === 9 && this.frames[this.currentFrame].length >= 1) {
```
Comment (maryrosecook) [inline src/Game.js]: Could the code use a constant, rather than `9`?

File: src/Game.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    this._currentFrameScore().push(score);
+    this._checkBonus(score);
+    if (this.frames[8][1] == 1) {
+      this.frames[8][0].push(score);
+      this.frames[this.currentFrame - 1][1] -= 1;
+    }
+  }
+};
+
+Game.prototype._checkFrame = function() {
```
Comment (maryrosecook) [inline src/Game.js]: Could `_checkFrame` have a name that more clearly conveys what it does?

File: src/Game.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    this._addFrame();
+  }
+};
+
+Game.prototype._addFrame = function() {
+  this.currentFrame += 1;
+  this.frames.push([[]]);
+};
+
+Game.prototype._checkBonus = function(score) {
```
Comment (maryrosecook) [inline src/Game.js]: It feels like `_checkBonus` is going to return a boolean, but it actually changes some data.  Could it be renamed?

File: src/Game.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+Game.prototype._applyBonus = function(score) {
+  for(var i = 0; i < this.frames.length; i++) {
+    if (this.frames[i][1] >= 1) {
+      this.frames[i][0].push(score);
+      this.frames[i][1] -= 1;
+    }
+  }
+};
+
+Game.prototype._currentFrameScore = function() {
```
Comment (maryrosecook) [inline src/Game.js]: Love this name!

File: src/Game.js (hunk 2 above)
Comment (maryrosecook) [inline src/Game.js]: Could line 7 be extracted to a method?

File: src/Game.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+Game.prototype.frameTotal = function() {
+ return this._currentFrameScore().reduce(function(a, b) {
+    return a + b;
+  }, 0);
+};
+
+Game.prototype.total = function() {
+  var total = 0;
+  for (var i = 0; i < this.frames.length; i++) {
+    total += this.frames[i][0].reduce(function(a, b) {
```
Comment (maryrosecook) [inline src/Game.js]: Love that you're using reduce!

File: src/Game.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+Game.prototype.frameTotal = function() {
+ return this._currentFrameScore().reduce(function(a, b) {
+    return a + b;
+  }, 0);
+};
+
+Game.prototype.total = function() {
+  var total = 0;
+  for (var i = 0; i < this.frames.length; i++) {
```
Comment (maryrosecook) [inline src/Game.js]: Having a separate frame object may clean up this code.  The Frame object can calculate the total for a frame and then you can reduce over that.

File: src/Game.js (hunk 5 above)
Comment (JG075) [inline src/Game.js]: Thanks for the helpful review!

**Comment by maryrosecook** [comment]
Sure thing!

--
https://maryrosecook.com

**Pull request closed without merging**